### PR TITLE
Migrate existing CI to GitHub Actions

### LINKED ISSUE #452: Migrate CI to GitHub Actions
Similar to https://github.com/jaegertracing/jaeger/issues/2645

* [x] migrate steps from .travis.yml to GHA workflows
* [x] update build badges in README
* [x] remove .travis.yml
* [x] change required checks to new GH jobs

### PR DESCRIPTION
**Pull request by tylerauerbeck**
## Which problem is this PR solving?

- Resolves #452 

## Short description of the changes

- This PR  takes the existing CI in `.travis.yml` and migrates it to use GH Actions.

### CODE REVIEW

File: .github/workflows/ci-unit-tests.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    - name: Run Linter
+      run: |
+        npm run lint
+      if: ${{ matrix.env.LINT == '1' }}
+    - name: Run Tests
+      run: |
+        make test-without-build
+    - name: Run Code Coverage
+      run: |
+        npm run cover && bash <(curl -s https://codecov.io/bash )
```
Comment (yurishkuro) [inline .github/workflows/ci-unit-tests.yml:57]: better to use codecov action, see https://github.com/jaegertracing/jaeger/blob/master/.github/workflows/ci-unit-tests.yml#L25

File: .github/workflows/ci-crossdock.yml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    - uses: actions/setup-node@v2
+      with:
+        node-version: "0.10"
+
+    - name: Run Crossdock
+      run: |
+        make crossdock-fresh
+        echo hello
+        docker images
+        echo hello
```
Comment (yurishkuro) [inline .github/workflows/ci-crossdock.yml:65]: debug logs?

**Comment by yurishkuro** [comment]
👏 👏 👏

**Comment by tylerauerbeck** [comment]
@yurishkuro These should be fixed now.

**Review by yurishkuro — APPROVED** [review]
Thanks!

In the future I recommend submitting PR not from master branch of your fork.

**Pull request merged**